AhoCorasick.py

```diff
@@ -0,0 +1,133 @@
+# implementing AhoCorasick using adjacency list

```

Comment:
please give a brief description about the algorithm. 
- max 4 lines
- one line should contain max 80 characters